PyStemmusScope/soil_io.py

```diff
@@ -0,0 +1,229 @@
+    ds = xr.combine_by_coords(datasets)
+
+    return ds
+
+
+def _read_lambda_coef(lambda_directory, lat, lon):
+    """Internal function that reads the lambda coefficient files and returns the data
+    of interest in a dictionary.
+
+    Args:
+        lambda_directory (Path): Path to the directory which contains the lambda data.
+        lat (float): Latitude of the site of interest (in degrees North)
+        lon (float): Longitude of the site of interest (in degrees East)
+
+    Returns:
+        dict: Dictionary containing the lambda coefficient data.
+    """
+    lambda_files = sorted(lambda_directory.glob("lambda_l*.nc"))
+
+    ds = _open_multifile_datasets(lambda_files, lat, lon)
+
+    # which depth indices the STEMMUS_SCOPE model expects
+    indices = [0, 2, 4, 5, 6, 7]
+    coef_lambda = ds['lambda'].isel(depth=indices).values
+
+    lambda_matfile = {'Coef_Lamda': coef_lambda}
+
+    return lambda_matfile
+
+
+def _read_soil_composition(soil_data_path, lat, lon):
+    """Internal function that reads the soil composition files and returns them in a
+    dictionary.
+
+    Args:
+        soil_data_path (Path): Path to the directory which contains the soil data.
+        lat (float): Latitude of the site of interest (in degrees North)
+        lon (float): Longitude of the site of interest (in degrees East)
+
+    Returns:
+        dict: Dictionary containing the soil composition data.
+    """
+    soil_comp_fnames = ['CLAY1.nc', 'CLAY2.nc', 'OC1.nc', 'OC2.nc', 'SAND1.nc',
+                        'SAND2.nc', 'SILT1.nc', 'SILT2.nc']
+
+    soil_comp_paths = [soil_data_path / fname for fname in soil_comp_fnames]
+
+    ds = _open_multifile_datasets(soil_comp_paths, lat, lon)
+
+    depths_indices = [0, 2, 4, 5, 6, 7]
+    ds = ds.isel(depth=depths_indices)
+
+    clay_fraction = vc.percent_to_fraction(ds['CLAY'].values)
+    sand_fraction = vc.percent_to_fraction(ds['SAND'].values)
+    organic_fraction = vc.hundredth_percent_to_fraction(ds['OC'].values)
+
+    soil_comp_matfiledata = {
+        'FOC': clay_fraction,
+        'FOS': sand_fraction,
+        'MSOC': organic_fraction,
+    }
+
+    return soil_comp_matfiledata
+
+
+def _read_hydraulic_parameters(soil_data_path, lat, lon):
+    """Internal function that reads the soil hydraulic parameters from the Schaap
+    dataset and returns them in a dictionary.
+
+    Args:
+        soil_data_path (Path): Path to the directory which contains the soil data.
+        lat (float): Latitude of the site of interest (in degrees North)
+        lon (float): Longitude of the site of interest (in degrees East)
+
+    Returns:
+        dict: Dictionary containing the hydraulic parameters.
+    """
+    ptf_files = sorted((soil_data_path / 'Schaap').glob('PTF_*.nc'))
+    ds = _open_multifile_datasets(
+        ptf_files, lat, lon, lat_key='latitude', lon_key='longitude'
+    )
+
+    depths = [0, 5, 30, 60, 100, 200]
```

Comment:
Let's  move `depths = [0, 5, 30, 60, 100, 200] ` to `prepare_soil_data` and here consider it as an input argument to the function ` _read_hydraulic_parameters()` as : `_read_hydraulic_parameters(soil_data_path, lat, lon, depths)`. We can add a `ToDo` to this module that the `depths` can be read from config file.